pkg/services/sqlstore/migrations/object_store_mig.go

```diff
@@ -0,0 +1,150 @@
+			{Name: "key", Type: migrator.DB_NVarchar, Length: 1024, Nullable: false, IsPrimaryKey: true},
+
+			// This is an optimization for listing everything at the same level in the object store
+			{Name: "parent_folder_key", Type: migrator.DB_NVarchar, Length: 1024, Nullable: false},
+
+			// The object type
+			{Name: "kind", Type: migrator.DB_NVarchar, Length: 255, Nullable: false},
+
+			// The raw object body (any byte array)
+			{Name: "body", Type: migrator.DB_Blob, Nullable: false},
+			{Name: "size", Type: migrator.DB_BigInt, Nullable: false},
+			{Name: "etag", Type: migrator.DB_NVarchar, Length: 32, Nullable: false}, // md5(body)
+			{Name: "version", Type: migrator.DB_NVarchar, Length: 128, Nullable: false},
+
+			// Who changed what when -- We should avoid JOINs with other tables in the database
+			{Name: "updated", Type: migrator.DB_DateTime, Nullable: false},
+			{Name: "created", Type: migrator.DB_DateTime, Nullable: false},
+			{Name: "updated_by", Type: migrator.DB_Int, Nullable: false},
+			{Name: "created_by", Type: migrator.DB_Int, Nullable: false},
+
+			// For objects that are synchronized from an external source (ie provisioning or git)
+			{Name: "sync_src", Type: migrator.DB_Text, Nullable: true},
+			{Name: "sync_time", Type: migrator.DB_DateTime, Nullable: true},
+
+			// Summary data (always extracted from the `body` column)
+			{Name: "name", Type: migrator.DB_NVarchar, Length: 255, Nullable: false},
+			{Name: "description", Type: migrator.DB_NVarchar, Length: 255, Nullable: true},
+			{Name: "labels", Type: migrator.DB_Text, Nullable: true}, // JSON object
+			{Name: "fields", Type: migrator.DB_Text, Nullable: true}, // JSON object
+			{Name: "errors", Type: migrator.DB_Text, Nullable: true}, // JSON object
+		},
+		PrimaryKeys: []string{"key"},
+		Indices: []*migrator.Index{
+			{Cols: []string{"parent_folder_key"}}, // list in folder
+			{Cols: []string{"kind"}},              // filter by type
+		},
+	})
+
+	tables = append(tables, migrator.Table{
+		Name: "object_labels",
+		Columns: []*migrator.Column{
+			{Name: "key", Type: migrator.DB_NVarchar, Length: 1024, Nullable: false},
+			{Name: "label", Type: migrator.DB_NVarchar, Length: 191, Nullable: false},
+			{Name: "value", Type: migrator.DB_NVarchar, Length: 1024, Nullable: false},
+		},
+		Indices: []*migrator.Index{
+			{Cols: []string{"key", "label"}, Type: migrator.UniqueIndex},
+		},
+	})
+
+	tables = append(tables, migrator.Table{
+		Name: "object_ref",
+		Columns: []*migrator.Column{
+			// Source:
+			{Name: "key", Type: migrator.DB_NVarchar, Length: 1024, Nullable: false},
+
+			// Address (defined in the body, not resolved, may be invalid and change)
+			{Name: "kind", Type: migrator.DB_NVarchar, Length: 255, Nullable: false},
+			{Name: "type", Type: migrator.DB_NVarchar, Length: 255, Nullable: true},
+			{Name: "uid", Type: migrator.DB_NVarchar, Length: 1024, Nullable: true},
+
+			// Runtime calcs (will depend on the system state)
```

Comment:
I'm not sure what this is :)

Reply:
This takes the internal references (eg: type:influx uid:ABC) and says if it resolves ok and to what